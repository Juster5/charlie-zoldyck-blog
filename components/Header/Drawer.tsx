import React from 'react'
import './index.scss'

type menuItem = {
  title?: string
  children: {
    title: ''
  }[]
}

type DrawerProps = {
  menu?: menuItem[]
}

const Drawer: React.FC<DrawerProps> = (props) => {
  return (
    <div className='collpase-wrapper'>
 
    </div>
  )
}

export default Drawer