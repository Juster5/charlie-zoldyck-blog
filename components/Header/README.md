### 头部导航栏组件