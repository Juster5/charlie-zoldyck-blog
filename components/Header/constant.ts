export const navs = [
  {
    title: 'Buy crypto',
    link: '/marker'
  },
  {
    title: 'Discover',
    link: '/marker'
  },
  {
    title: 'Trade',
    link: '/marker'
  },
  {
    title: 'Grow',
    link: '/marker'
  },
  {
    title: 'Build',
    link: '/marker'
  },
  {
    title: 'Learn',
    link: '/marker'
  },
  {
    title: 'More',
    link: '/marker'
  }
]