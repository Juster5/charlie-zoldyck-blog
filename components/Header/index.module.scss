.header {
  display: flex;
  align-items: center;
  padding: 0 24px;
  height: 48px;
  background-color: #121212;
  color: #fff;
}
.navs {
  display: flex;
  align-items: center;  
  flex: 1;
  .nav-item {
    display: flex;
    padding: 0 10px;
    align-items: center;
    height: 100%;
    cursor: pointer;
  }  
}

.logo-wrapper {
  margin-right: 24px;
}

.login-wrapper {
  display: flex;
  align-items: center;
  .login {
    display: inline-block;
    margin-right: 12px;
    cursor: pointer;
  }
  .signup{
    display: inline-block;
    padding: 6px 8px;
    border: 1px solid #fff;
    border-radius: 4px;
    cursor: pointer;
  }
}
.languages {
  margin-left:12px;
  font-size: 20px;
  cursor: pointer;
  &::before{
    content: "\e039";
    font-family: fa-f5350cc488;  
  } 
}