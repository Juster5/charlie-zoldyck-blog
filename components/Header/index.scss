@import '@/styles/media.scss';

.okx-header {
  z-index: 9999;
  display: flex;
  align-items: center;
  height: 48px;
  padding: 0 24px;
  background-color: #121212;
  color: #fff;

  @include responsive(sm) {
    position: fixed;
    top: 0;
    left: 0;
    width: 100%;
    padding: 0 16px;
  }

  .navs {
    display: flex;
    align-items: center;
    flex: 1;
    height: 100%;
    .nav-item {
      position: relative;
      display: flex;
      padding: 0 10px;
      align-items: center;
      height: 100%;
      cursor: pointer;
      &:hover {
        .title-arrow {
          transform: rotate(-180deg);
          margin-left: 12px;
        }
      }
    }
    .title-arrow {
      margin-left: 12px;
      transition: all 0.1s ease;
    }
  }

  .logo-wrapper {
    margin-right: 24px;
    font-size: 0;
    cursor: pointer;
  }

  .languages {
    display: flex;
    align-items: center;
    margin-left: 12px;
    font-size: 20px;
    cursor: pointer;
    height: 100%;
  }
}

.login-wrapper {
  display: flex;
  align-items: center;
  .mr12 {
    margin-right: 12px;
  }

  .login {
    display: inline-block;
    cursor: pointer;
  }
  .signup {
    display: inline-block;
    padding: 6px 8px;
    border: 1px solid #fff;
    border-radius: 4px;
    cursor: pointer;
    @include responsive(sm) {
      padding: 4px 6px;
    }
  }
  .menu {
    margin-left: 20px;
    font-size: 20px;
  }
}

.fixed-block {
  display: none;
  @include responsive(sm) {
    display: block;
  }
}

.drawer-wrapper {
  z-index: 9999;
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  padding: 48px 24px;
  background-color: #000;
  overflow: auto;
  color: #fafafa;
  transform: translateX(100%);
  transition: all 0.5s ease;

  &.show {
    transform: translateX(0);
  }
  @include responsive(bg) {
    display: none;
  }

  .okx-header-footer-close {
    position: absolute;
    right: 24px;
    top: 20px;
    font-size: 20px;
  }

  .mobile-language {
    margin: 16px 0;
    .mobile-lang-btn {
      position: relative;
      z-index: 9999;
      display: inline-block;
      width: 48%;
      background: #242424;
      border-radius: 4px;
      margin-bottom: 16px;
      padding: 10px 16px;
      text-align: center;
      &:first-child {
        margin-right: 4%;
      }
      &.active {
        border: 1px solid #fff;
      }
    }
  }
}
