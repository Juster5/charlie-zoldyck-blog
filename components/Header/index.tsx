import type { NextPage } from 'next';
import Image from 'next/image';
import './index.scss'
import CollpaseMenu from '../CollapseMenu';

import { navs } from './constant';

const Header: NextPage = () => {
  return (
    <header className="okx-header">
      {/* logo区域 */}
      <div className="logo-wrapper">
        <Image src="/images/logo.png" alt="OKX" width={82} height={36} />
      </div>

      {/* 导航栏区域 */}
      <div className="navs">
        {
          navs.map(el=>{
            return (
              <CollpaseMenu menu={el.children} key={el.title}>
                <div className="nav-item sm-screen-hidden">{el.title}</div>              
              </CollpaseMenu>
            )
          })
        }
      </div>

      {/* 登录区域 */}
      <div className="login-wrapper">
        <span className="login sm-screen-hidden">Log in</span>
        <span className="signup">Sign up</span>
        <span className="menu bg-screen-hidden"></span>
      </div>

      {/* 多语言区域 */}
      <CollpaseMenu position="right" 
        menu={[{
          title: '简体中文'
        }, {
          title: 'English'
        }]}
      >
        <div className="languages sm-screen-hidden" >
          <span className='okx-header-footer-language'></span>     
        </div>
      </CollpaseMenu>
    </header>    
  )
}

export default Header