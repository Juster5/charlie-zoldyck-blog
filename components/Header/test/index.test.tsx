import { render, screen } from '@testing-library/react'
import Header from '..'

// 渲染测试
describe('header组件渲染测试', () => {
  test('logo渲染', () => {
    render(<Header />)
    const okxImg = screen.getByRole('img', {
      name: 'OKX',
    })
    expect(okxImg).toBeInTheDocument()
  })

  test('导航渲染', async () => {
    render(<Header />)
    const menuItems = screen.queryAllByText('Buy crypto')
    expect(menuItems[0]).toBeInTheDocument()
  })
})
