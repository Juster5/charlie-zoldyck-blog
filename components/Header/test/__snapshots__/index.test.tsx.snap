// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div>
  <header
    class="okx-header"
  >
    <div
      class="logo-wrapper"
    >
      <img
        alt="OKX"
        data-nimg="1"
        decoding="async"
        height="36"
        loading="lazy"
        src="/_next/image?url=%2Fimages%2Flogo.png&w=256&q=75"
        srcset="/_next/image?url=%2Fimages%2Flogo.png&w=96&q=75 1x, /_next/image?url=%2Fimages%2Flogo.png&w=256&q=75 2x"
        style="color: transparent;"
        width="82"
      />
    </div>
    <div
      class="navs"
    >
      <div
        class="collpase-wrapper"
      >
        <div
          class="nav-item sm-screen-hidden"
        >
          <span>
            header_buy_crypto
          </span>
          <span
            class="title-arrow okx-header-footer-arrow-chevrons-down"
          />
        </div>
        <div
          class="collpase-menu collpase-center"
        >
          <div
            class="menu-item"
          >
            <div
              class="menu-item__icon  okx-header-footer-buy-with-card"
            />
            <div
              class="menu-item__text"
            >
              <div
                class="title"
              >
                header_buy_with_card
              </div>
              <div
                class="sub-title"
              >
                header_buy_with_other
              </div>
            </div>
          </div>
          <div
            class="menu-item"
          >
            <div
              class="menu-item__icon  okx-header-footer-p2p-trading"
            />
            <div
              class="menu-item__text"
            >
              <div
                class="title"
              >
                header_p2p
              </div>
              <div
                class="sub-title"
              >
                header_p2p_rate
              </div>
            </div>
          </div>
          <div
            class="menu-item"
          >
            <div
              class="menu-item__icon  okx-header-footer-third-party-payment"
            />
            <div
              class="menu-item__text"
            >
              <div
                class="title"
              >
                header_third_pay
              </div>
              <div
                class="sub-title"
              >
                header_third_pay_type
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="collpase-wrapper"
      >
        <div
          class="nav-item sm-screen-hidden"
        >
          <span>
            header_discover
          </span>
          <span
            class="title-arrow okx-header-footer-arrow-chevrons-down"
          />
        </div>
        <div
          class="collpase-menu collpase-center"
        >
          <div
            class="menu-item"
          >
            <div
              class="menu-item__icon  okx-header-footer-buy-with-card"
            />
            <div
              class="menu-item__text"
            >
              <div
                class="title"
              >
                header_buy_with_card
              </div>
              <div
                class="sub-title"
              >
                header_buy_with_other
              </div>
            </div>
          </div>
          <div
            class="menu-item"
          >
            <div
              class="menu-item__icon  okx-header-footer-p2p-trading"
            />
            <div
              class="menu-item__text"
            >
              <div
                class="title"
              >
                header_p2p
              </div>
              <div
                class="sub-title"
              >
                header_p2p_rate
              </div>
            </div>
          </div>
          <div
            class="menu-item"
          >
            <div
              class="menu-item__icon  okx-header-footer-third-party-payment"
            />
            <div
              class="menu-item__text"
            >
              <div
                class="title"
              >
                header_third_pay
              </div>
              <div
                class="sub-title"
              >
                header_third_pay_type
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="collpase-wrapper"
      >
        <div
          class="nav-item sm-screen-hidden"
        >
          <span>
            header_trade
          </span>
          <span
            class="title-arrow okx-header-footer-arrow-chevrons-down"
          />
        </div>
        <div
          class="collpase-menu collpase-center"
        >
          <div
            class="menu-item"
          >
            <div
              class="menu-item__icon  okx-header-footer-buy-with-card"
            />
            <div
              class="menu-item__text"
            >
              <div
                class="title"
              >
                header_buy_with_card
              </div>
              <div
                class="sub-title"
              >
                header_buy_with_other
              </div>
            </div>
          </div>
          <div
            class="menu-item"
          >
            <div
              class="menu-item__icon  okx-header-footer-p2p-trading"
            />
            <div
              class="menu-item__text"
            >
              <div
                class="title"
              >
                header_p2p
              </div>
              <div
                class="sub-title"
              >
                header_p2p_rate
              </div>
            </div>
          </div>
          <div
            class="menu-item"
          >
            <div
              class="menu-item__icon  okx-header-footer-third-party-payment"
            />
            <div
              class="menu-item__text"
            >
              <div
                class="title"
              >
                header_third_pay
              </div>
              <div
                class="sub-title"
              >
                header_third_pay_type
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="collpase-wrapper"
      >
        <div
          class="nav-item sm-screen-hidden"
        >
          <span>
            header_grow
          </span>
          <span
            class="title-arrow okx-header-footer-arrow-chevrons-down"
          />
        </div>
        <div
          class="collpase-menu collpase-center"
        >
          <div
            class="menu-item"
          >
            <div
              class="menu-item__icon  okx-header-footer-buy-with-card"
            />
            <div
              class="menu-item__text"
            >
              <div
                class="title"
              >
                header_buy_with_card
              </div>
              <div
                class="sub-title"
              >
                header_buy_with_other
              </div>
            </div>
          </div>
          <div
            class="menu-item"
          >
            <div
              class="menu-item__icon  okx-header-footer-p2p-trading"
            />
            <div
              class="menu-item__text"
            >
              <div
                class="title"
              >
                header_p2p
              </div>
              <div
                class="sub-title"
              >
                header_p2p_rate
              </div>
            </div>
          </div>
          <div
            class="menu-item"
          >
            <div
              class="menu-item__icon  okx-header-footer-third-party-payment"
            />
            <div
              class="menu-item__text"
            >
              <div
                class="title"
              >
                header_third_pay
              </div>
              <div
                class="sub-title"
              >
                header_third_pay_type
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="collpase-wrapper"
      >
        <div
          class="nav-item sm-screen-hidden"
        >
          <span>
            header_build
          </span>
          <span
            class="title-arrow okx-header-footer-arrow-chevrons-down"
          />
        </div>
        <div
          class="collpase-menu collpase-center"
        >
          <div
            class="menu-item"
          >
            <div
              class="menu-item__icon  okx-header-footer-buy-with-card"
            />
            <div
              class="menu-item__text"
            >
              <div
                class="title"
              >
                header_buy_with_card
              </div>
              <div
                class="sub-title"
              >
                header_buy_with_other
              </div>
            </div>
          </div>
          <div
            class="menu-item"
          >
            <div
              class="menu-item__icon  okx-header-footer-p2p-trading"
            />
            <div
              class="menu-item__text"
            >
              <div
                class="title"
              >
                header_p2p
              </div>
              <div
                class="sub-title"
              >
                header_p2p_rate
              </div>
            </div>
          </div>
          <div
            class="menu-item"
          >
            <div
              class="menu-item__icon  okx-header-footer-third-party-payment"
            />
            <div
              class="menu-item__text"
            >
              <div
                class="title"
              >
                header_third_pay
              </div>
              <div
                class="sub-title"
              >
                header_third_pay_type
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="collpase-wrapper"
      >
        <div
          class="nav-item sm-screen-hidden"
        >
          <span>
            header_learn
          </span>
          <span
            class="title-arrow okx-header-footer-arrow-chevrons-down"
          />
        </div>
        <div
          class="collpase-menu collpase-center"
        >
          <div
            class="menu-item"
          >
            <div
              class="menu-item__icon  okx-header-footer-buy-with-card"
            />
            <div
              class="menu-item__text"
            >
              <div
                class="title"
              >
                header_buy_with_card
              </div>
              <div
                class="sub-title"
              >
                header_buy_with_other
              </div>
            </div>
          </div>
          <div
            class="menu-item"
          >
            <div
              class="menu-item__icon  okx-header-footer-p2p-trading"
            />
            <div
              class="menu-item__text"
            >
              <div
                class="title"
              >
                header_p2p
              </div>
              <div
                class="sub-title"
              >
                header_p2p_rate
              </div>
            </div>
          </div>
          <div
            class="menu-item"
          >
            <div
              class="menu-item__icon  okx-header-footer-third-party-payment"
            />
            <div
              class="menu-item__text"
            >
              <div
                class="title"
              >
                header_third_pay
              </div>
              <div
                class="sub-title"
              >
                header_third_pay_type
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="collpase-wrapper"
      >
        <div
          class="nav-item sm-screen-hidden"
        >
          <span>
            header_more
          </span>
          <span
            class="title-arrow okx-header-footer-arrow-chevrons-down"
          />
        </div>
        <div
          class="collpase-menu collpase-center"
        >
          <div
            class="menu-item"
          >
            <div
              class="menu-item__icon  okx-header-footer-buy-with-card"
            />
            <div
              class="menu-item__text"
            >
              <div
                class="title"
              >
                header_buy_with_card
              </div>
              <div
                class="sub-title"
              >
                header_buy_with_other
              </div>
            </div>
          </div>
          <div
            class="menu-item"
          >
            <div
              class="menu-item__icon  okx-header-footer-p2p-trading"
            />
            <div
              class="menu-item__text"
            >
              <div
                class="title"
              >
                header_p2p
              </div>
              <div
                class="sub-title"
              >
                header_p2p_rate
              </div>
            </div>
          </div>
          <div
            class="menu-item"
          >
            <div
              class="menu-item__icon  okx-header-footer-third-party-payment"
            />
            <div
              class="menu-item__text"
            >
              <div
                class="title"
              >
                header_third_pay
              </div>
              <div
                class="sub-title"
              >
                header_third_pay_type
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="login-wrapper"
    >
      <span
        class="login sm-screen-hidden mr12"
      >
        common_login
      </span>
      <span
        class="signup"
      >
        common_signup
      </span>
      <span
        class="menu bg-screen-hidden okx-header-footer-hamburger"
      />
    </div>
    <div
      class="collpase-wrapper"
    >
      <div
        class="languages sm-screen-hidden"
      >
        <span
          class="okx-header-footer-language"
        />
      </div>
      <div
        class="collpase-menu collpase-right"
      >
        <div
          class="menu-item"
        >
          <div
            class="menu-item__text"
          >
            <div
              class="title"
            >
              English
            </div>
            <div
              class="sub-title"
            />
          </div>
        </div>
        <div
          class="menu-item"
        >
          <div
            class="menu-item__text"
          >
            <div
              class="title"
            >
              简体中文
            </div>
            <div
              class="sub-title"
            />
          </div>
        </div>
      </div>
    </div>
  </header>
  <div
    class="drawer-wrapper "
  >
    <div
      class="okx-header-footer-close"
    />
    <div
      class="login-wrapper"
    >
      <span
        class="signup mr12"
      >
        common_signup
      </span>
      <span
        class="login"
      >
        common_login
      </span>
    </div>
    <div
      class="mobile-language"
    >
      <div
        class="mobile-lang-btn active"
      >
        English
      </div>
      <div
        class="mobile-lang-btn "
      >
        简体中文
      </div>
    </div>
    <ul
      class="okx-dropdown-menu"
    >
      <li
        class=""
      >
        <div
          class="menu-item"
        >
          <div
            class="menu-item-title"
          >
            header_buy_crypto
          </div>
          <div
            class="menu-arrow okx-header-footer-arrow-chevrons-down"
          />
        </div>
        <div
          class="sub-item-wrapper"
        >
          <div
            class="sub-item"
          >
            header_buy_with_other
          </div>
          <div
            class="sub-item"
          >
            header_p2p_rate
          </div>
          <div
            class="sub-item"
          >
            header_third_pay_type
          </div>
        </div>
      </li>
      <li
        class=""
      >
        <div
          class="menu-item"
        >
          <div
            class="menu-item-title"
          >
            header_discover
          </div>
          <div
            class="menu-arrow okx-header-footer-arrow-chevrons-down"
          />
        </div>
        <div
          class="sub-item-wrapper"
        >
          <div
            class="sub-item"
          >
            header_buy_with_other
          </div>
          <div
            class="sub-item"
          >
            header_p2p_rate
          </div>
          <div
            class="sub-item"
          >
            header_third_pay_type
          </div>
        </div>
      </li>
      <li
        class=""
      >
        <div
          class="menu-item"
        >
          <div
            class="menu-item-title"
          >
            header_trade
          </div>
          <div
            class="menu-arrow okx-header-footer-arrow-chevrons-down"
          />
        </div>
        <div
          class="sub-item-wrapper"
        >
          <div
            class="sub-item"
          >
            header_buy_with_other
          </div>
          <div
            class="sub-item"
          >
            header_p2p_rate
          </div>
          <div
            class="sub-item"
          >
            header_third_pay_type
          </div>
        </div>
      </li>
      <li
        class=""
      >
        <div
          class="menu-item"
        >
          <div
            class="menu-item-title"
          >
            header_grow
          </div>
          <div
            class="menu-arrow okx-header-footer-arrow-chevrons-down"
          />
        </div>
        <div
          class="sub-item-wrapper"
        >
          <div
            class="sub-item"
          >
            header_buy_with_other
          </div>
          <div
            class="sub-item"
          >
            header_p2p_rate
          </div>
          <div
            class="sub-item"
          >
            header_third_pay_type
          </div>
        </div>
      </li>
      <li
        class=""
      >
        <div
          class="menu-item"
        >
          <div
            class="menu-item-title"
          >
            header_build
          </div>
          <div
            class="menu-arrow okx-header-footer-arrow-chevrons-down"
          />
        </div>
        <div
          class="sub-item-wrapper"
        >
          <div
            class="sub-item"
          >
            header_buy_with_other
          </div>
          <div
            class="sub-item"
          >
            header_p2p_rate
          </div>
          <div
            class="sub-item"
          >
            header_third_pay_type
          </div>
        </div>
      </li>
      <li
        class=""
      >
        <div
          class="menu-item"
        >
          <div
            class="menu-item-title"
          >
            header_learn
          </div>
          <div
            class="menu-arrow okx-header-footer-arrow-chevrons-down"
          />
        </div>
        <div
          class="sub-item-wrapper"
        >
          <div
            class="sub-item"
          >
            header_buy_with_other
          </div>
          <div
            class="sub-item"
          >
            header_p2p_rate
          </div>
          <div
            class="sub-item"
          >
            header_third_pay_type
          </div>
        </div>
      </li>
      <li
        class=""
      >
        <div
          class="menu-item"
        >
          <div
            class="menu-item-title"
          >
            header_more
          </div>
          <div
            class="menu-arrow okx-header-footer-arrow-chevrons-down"
          />
        </div>
        <div
          class="sub-item-wrapper"
        >
          <div
            class="sub-item"
          >
            header_buy_with_other
          </div>
          <div
            class="sub-item"
          >
            header_p2p_rate
          </div>
          <div
            class="sub-item"
          >
            header_third_pay_type
          </div>
        </div>
      </li>
    </ul>
  </div>
</div>
`;

exports[`header组件渲染测试 logo渲染 1`] = `<div />`;
